File: app/src/main/java/io/github/sithengineer/motoqueiro/data/sync/RemoteSyncService.java
package io.github.sithengineer.motoqueiro.data.sync;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;

public class RemoteSyncService extends Service {

  private static final Object _lock = new Object();

  private static RemoteSyncAdapter adapter = null;

  @Override public void onCreate() {
    super.onCreate();
    synchronized (_lock) {
      if (adapter == null) {
        adapter = new RemoteSyncAdapter(getApplicationContext(), true, false);
      }
    }
  }

  @Nullable @Override public IBinder onBind(Intent intent) {
    return adapter.getSyncAdapterBinder();
  }
}

File: app/src/main/java/io/github/sithengineer/motoqueiro/data/sync/RemoteSyncService.kt
package io.github.sithengineer.motoqueiro.data.sync

import android.app.Service
import android.content.Intent
import android.os.IBinder

class RemoteSyncService : Service() {

  private val _lock = Any()

  override fun onCreate() {
    super.onCreate()
    synchronized(_lock) {
      if (adapter == null) {
        adapter = RemoteSyncAdapter(applicationContext, true, false)
      }
    }
  }

  override fun onBind(intent: Intent): IBinder? {
    return adapter!!.syncAdapterBinder
  }

  companion object {
    private var adapter: RemoteSyncAdapter? = null
  }
}
